diff --git a/wikisql/lib/dbengine.py b/wikisql/lib/dbengine.py
index 83eeb9a..e7f7323 100644
--- a/wikisql/lib/dbengine.py
+++ b/wikisql/lib/dbengine.py
@@ -14,6 +14,7 @@ class DBEngine:
 
     def __init__(self, fdb):
         self.db = records.Database('sqlite:///{}'.format(fdb))
+        self.conn = self.db.get_connection()
 
     def execute_query(self, table_id, query, *args, **kwargs):
         return self.execute(table_id, query.sel_index, query.agg_index, query.conditions, *args, **kwargs)
@@ -21,7 +22,7 @@ class DBEngine:
     def execute(self, table_id, select_index, aggregation_index, conditions, lower=True):
         if not table_id.startswith('table'):
             table_id = 'table_{}'.format(table_id.replace('-', '_'))
-        table_info = self.db.query('SELECT sql from sqlite_master WHERE tbl_name = :name', name=table_id).all()[0].sql
+        table_info = self.conn.query('SELECT sql from sqlite_master WHERE tbl_name = :name', name=table_id).all()[0].sql
         schema_str = schema_re.findall(table_info)[0]
         schema = {}
         for tup in schema_str.split(', '):
@@ -47,5 +48,5 @@ class DBEngine:
         if where_clause:
             where_str = 'WHERE ' + ' AND '.join(where_clause)
         query = 'SELECT {} AS result FROM {} {}'.format(select, table_id, where_str)
-        out = self.db.query(query, **where_map)
-        return [o.result for o in out]
+        out = self.conn.query(query, **where_map)
+        return [o.result for o in out]
\ No newline at end of file
